evalai/utils/requests.py

```diff
@@ -105,3 +109,16 @@ def make_request(path, method, files=None, data=None):
     elif method == "DELETE":
         # TODO: Add support for DELETE request
         pass
+
+
+def check_compatibility(response):
+    if "Minimum-CLI-Version" in response.headers:
+        version = pkg_resources.require("evalai")[0].version
+        if semver.compare(version, response.headers["Minimum-CLI-Version"]) is -1:
+            echo(
+                style(
+                    "\nYour CLI ({}) is too old and needs to be updated\n".format(version),
+                    fg="red",
+                    bold=True,
+                )
+            )
```

Comment:
Please add statement to exit with non-zero exit code, as it will continue to execute the command even if the CLI version is to not compatible with Server.